waspc/libs/auth/src/sdk/password.ts

```diff
@@ -0,0 +1,47 @@
+import { hash, verify, type Options } from "@node-rs/argon2";
+
+// The options are the same as the ones used in the oslo/password library
+const hashingOptions: Options = {
+  memoryCost: 19456,
+  timeCost: 2,
+  outputLen: 32,
+  parallelism: 1,
+};
+
+/**
+ * Hashes a password using the Argon2 algorithm.
+ * - It normalizes the password password before hashing to ensure consistency.
+ * - It doesn't require a salt input as the argon2 library generates a random salt for each hash.
+ *
+ * @param password The password to hash
+ * @returns A PHC string (e.g. $argon2id$v=19$m=19456,t=2,p=1$...) whichs contains the hashed password and the salt.
+ * @see https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md for more information about the PHC string format.
+ */
+export async function hashPassword(password: string): Promise<string> {
+  return hash(normalizePassword(password), hashingOptions);
+}
+
+/**
+ *
+ * @param hashedPassword A PHC string (e.g. $argon2id$v=19$m=19456,t=2,p=1$...) whichs contains the hashed password and the salt.
+ * @param password The password to verify
+ * @throws An error if the password is invalid
+ */
```

Comment:
Formatting in parts of the JS doc seems off (sometimes there's a stop at the end, sometimes there isn't etc.)

Since we're writing the docs, let's write the best version we can:
- Perhaps a link to the PHC specification in this signature too.
- "The password to verify against the PHC string" is perhaps more informative.

I won't dig in fruther. The TL;DR is "please polish where possible" :)